File: examples/canonical-azure-functions/blob-trigger-receipt-ocr/python/tests/smoke/test_handler.py
import unittest

from examples.canonical_azure_functions_import import load_python_handler


handler = load_python_handler(
    "examples/canonical-azure-functions/blob-trigger-receipt-ocr/python/src/handler.py"
)


class HandlerSmokeTest(unittest.TestCase):
    def test_receipt_blob_maps_to_cosmos_and_event_grid(self):
        result = handler.process_blob(
            {
                "name": "receipts/2026/05/06/r-1.pdf",
                "version": "v1",
                "content_type": "application/pdf",
                "bytes_len": 512,
                "merchant": "Contoso Market",
                "total": 42.37,
            }
        )
        self.assertEqual(result["cosmos_document"]["merchant"], "Contoso Market")
        self.assertEqual(result["event_grid_event"]["eventType"], "accb.receipt.ocr.completed")


if __name__ == "__main__":
    unittest.main()

File: examples/canonical-azure-functions/blob-trigger-receipt-ocr/typescript/tests/smoke/test_handler.ts
import assert from "node:assert/strict";
import { processBlob } from "../../src/handler";

const result = processBlob({ name: "receipts/r-1.pdf", version: "etag-1", merchant: "Contoso", total: 19.95 });
assert.equal(result.cosmos_document.blob_name, "receipts/r-1.pdf");
